src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -140,6 +140,9 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         foreach ($tokens as $token) {
             if ('variable' === $token[0]) {
                 if (!$optional || !array_key_exists($token[3], $defaults) || null !== $mergedParams[$token[3]] && (string) $mergedParams[$token[3]] !== (string) $defaults[$token[3]]) {
+                    if ($this->hasLookAround($token[2])) {
+                        $token[2] = $this->removeLookAround($token[2]);
```

Comment:
I would not override the value in the token array. This might have side-effects in the future if we use it for other stuff than the validation. Use a local variable instead